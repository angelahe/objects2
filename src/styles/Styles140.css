.AppContainer {
    background-color: #e6f5ff;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;

}

.AppPanel {
    background-color: whitesmoke;
    border-radius: 10px;
    border-width: 2px;
    border-color: #1aa3ff;
    margin: 10px;
    height: auto;
    width: 40%;
    justify-content: flex-start;

}
.Header {
    background-color: #1aa3ff;
    border-radius: 10px;
}

.AppBtn {
    color: white;
    height: 35px;
    width:35px;
    padding:0px;
    align-content: flex-start;
    border-radius: 5px;
    margin-left: 10px;

}
.ActionBox {
    height: auto;
    width: 100%;
    border-radius: 10px;

}
.ItemBox {
    border-radius: 10px;
    padding: 10px;
}
.AppHeader {
    background-color: #1aa3ff;
    display:flex;
    justify-content:space-between;
}

.AppList {
    width: 96%;
    padding: 5px;
}

.AppListItem {
    background-color: #e6f5ff;
}

.AppListItem:active {
    background-color: #ccebff;
    justify-content: space-around;
}

.AddItem {
    /*width: 50%;*/
    font-weight: bold;
    font-size: 1.5em;
    justify-content: flex-start;
    color: white;
    background-color: #1aa3ff;
    text-align: left;

}
.btnImg {
    height: 35px;
    width: 35px;
    background-color: #1aa3ff;
    margin:0;
}

.btnText {
    height: 35px;
    width: 200px;
    background-color: #1aa3ff;
    color: white;
    font-size: 1.5em;
    border-radius: 10px;

}

.DetailText {
    color: #1aa3ff;
    font-size: 1.5em;
    align-items: flex-start;
    justify-content: left;
    text-align: left;
    line-height: 1.5em;
}

.InputText {
    color: #1aa3ff;
    font-size: 1.5em;
}

.SummaryText {
    color: #1aa3ff;
    font-size: 1em;
    align-items: flex-start;
    text-align: left;
    margin-left: 10px;
    line-height: 1.5em;
}

hr {
    color: darkblue;
    height: 2px;
}

h2 {
    padding: 1px;
    color: #1aa3ff;
}

