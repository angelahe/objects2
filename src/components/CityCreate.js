import React from 'react';
import closebtn from '../images/close_FFFFFF.png';
import "../styles/Styles140.css";

const CityCreate = (props) => {

  console.log("props is ", props);

  function handleCloseClick() {
    console.log("in handleCloseClick");
  }

  function handleAddClick() {
    console.log("in handleAddClick function");
    const name = document.getElementById("cityName").value;
    const latitude = document.getElementById("latitude").value;
    const longitude = document.getElementById("longitude").value;
    const population = Number(document.getElementById("population").value);
    props.createClicked(name, latitude, longitude, population);
  }

  return(
    <div>
      <div className = "ItemBox AppHeader">
        <span className= "AddItem">Add City</span>
        <button className="AppBtn" onClick={handleCloseClick}>
          <img className="btnImg" src={closebtn} alt="Close"/>
        </button>
      </div>
      <div className = "ItemBox">
        <span className = "DetailText">City Name:</span>
        <input className="InputText" id="cityName" autoFocus={true}></input><br/>
        <span className = "DetailText">Latitude:</span>
        <input className="InputText" id="latitude" autoFocus={true}></input><br/>
        <span className = "DetailText">Longitude:</span>
        <input className="InputText" id="longitude" autoFocus={true}></input><br/>
        <span className = "DetailText">Population:</span>
        <input className = "InputText" id="population" type="number"></input><br/><br/>
        <button className="btnText" onClick={handleAddClick}>Add City</button>

      </div>
    </div>
  )
}
export default CityCreate