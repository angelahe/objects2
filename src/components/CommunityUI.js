import React from 'react';
import community from './community';
import CityCreate from './CityCreate';
import CityEdit from './CityEdit';
import CityDelete from './CityDelete';
import OneCommunity from './OneCommunity';
import CommunitySummary from './CommunitySummary';
import CityActions from './CityActions';
import addbtn from '../images/add_FFFFFF.png';

import '../styles/Styles140.css';

class CommunityUI extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      CommunityList: new community(),
      currentCity: null,
      addShow: false,
      editShow: false,
      actionsShow: false,
      deleteShow: false
    }
  }

  onBtnCloseClick = (closeType) => {
    if (closeType === "create") this.setState ({addShow: false});
    if (closeType === "edit") this.setState ({editShow: false});
    if (closeType === "delete") this.setState ({deleteShow: false});
    if (closeType === "actions") this.setState ({actionsShow: false});
  }

  onBtnAddClick = () => {
    this.setState({addShow: true});
  }

  onBtnEditClick = () => {
    this.setState({editShow: true});
  }

  onCityClick = (city) => {
    this.setState({
      actionsShow: true,
      currentCity: city
    })
  }

  onEditClick = (city) => {
    this.setState({
      editShow: true,
      currentCity: city
    })
  }

  onDeleteClick = (city) => {
    this.setState({
      deleteShow: true,
      currentCity: city
    })
  }

  onCityDelete = (city) => {
    this.state.CommunityList.deleteCity(city);

    this.setState({
      CommunityList: this.state.CommunityList,
      deleteShow: false
    })
  }

  onCityUpdate = (city, updated) => {
    console.log("in city update");
    this.state.CommunityList.updateCommunity(city, updated);

    this.setState({
      CommunityList: this.state.CommunityList,
      editShow: false
    })
  }


  onCityCreate = (name, latitude, longitude, population) => {
    this.state.CommunityList.addCommunity(name, latitude, longitude, population);

    this.setState({
      CommunityList: this.state.CommunityList,
      addShow: false
      }
    )

  }

  onMovedIn = (city, number) => {
    console.log("in moved in number is", number);
    this.state.CommunityList.movedIn(city, number);

    this.setState({
      CommunityList: this.state.CommunityList,
      actionsShow: false,
    });
  }

  onMovedOut = (city, number) => {
    console.log("in moved out number is", number);
    this.state.CommunityList.movedOut(city, number);

    this.setState({
      CommunityList: this.state.CommunityList,
      actionsShow: false,
    });
  }

  render() {
    const communities = this.state.CommunityList.Communities.map((city, index) =>
    <OneCommunity key = {city.Name}
                  city = {city}
                  cityClicked={this.onCityClick}
                  editClicked={this.onEditClick}
                  deleteClicked={this.onDeleteClick}
    />
    );

    return(
      <div>

        <h1>This is the Community UI</h1>
        <div className="AppContainer">
          <CommunitySummary communities={this.state.CommunityList}/>
          <div className="AppPanel">
            <div className="ItemBox AppHeader">
              <span className="AddItem">Add City</span>
              <button className="AppBtn" onClick={this.onBtnAddClick}>
                <img className="btnImg" src={addbtn} alt="Add"/>
              </button>
            </div>
            <div className = "AppList">
              {communities}
            </div>
          </div>
          <div className = "AppPanel">
            {(this.state.addShow)
              ? <CityCreate createClicked = {this.onCityCreate}
                            closeClicked={this.onBtnCloseClick}
                />
              :null
            }
            {(this.state.editShow)
              ? <CityEdit
                         city={this.state.currentCity}
                         updateCity = {this.onCityUpdate}
                         closeClicked={this.onBtnCloseClick}
                />
              : null
              }
            {(this.state.deleteShow)
              ? <CityDelete
                        city={this.state.currentCity}
                        deleteCity={this.onCityDelete}
                        closeClicked={this.onBtnCloseClick}
                />
              :null
              }
            {(this.state.actionsShow)
              ? <CityActions
                city={this.state.currentCity}
                movedIn={this.onMovedIn}
                movedOut={this.onMovedOut}
              />
              : null
            }
          </div>

        </div>
      </div>
    )
  }
}
export default CommunityUI
