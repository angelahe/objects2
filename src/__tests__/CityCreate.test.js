import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CityCreate from "../components/CityCreate";
import '../styles/Styles140.css'
import renderer from "react-test-renderer";
import CommunityUI from "../components/CommunityUI";

const props = new CommunityUI();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CityCreate createClicked = {props.onCityCreate}
                              closeClicked={props.onBtnCloseClick}
  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<CityCreate createClicked = {props.onCityCreate}
                      closeClicked={props.onBtnCloseClick}
  />);
});

test('CityActions renders correctly', () => {
  const tree = renderer.create(<CityCreate createClicked = {props.onCityCreate}
                                           closeClicked={props.onBtnCloseClick}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});