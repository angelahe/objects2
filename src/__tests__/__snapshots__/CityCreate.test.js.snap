// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityActions renders correctly 1`] = `
<div>
  <div
    className="ItemBox AppHeader"
  >
    <span
      className="AddItem"
    >
      Add City
    </span>
    <button
      className="AppBtn"
      onClick={[Function]}
    >
      <img
        alt="Close"
        className="btnImg"
        src="close_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="ItemBox"
  >
    <span
      className="DetailText"
    >
      City Name:
    </span>
    <input
      autoFocus={true}
      className="InputText"
      id="cityName"
    />
    <br />
    <span
      className="DetailText"
    >
      Latitude:
    </span>
    <input
      autoFocus={true}
      className="InputText"
      id="latitude"
    />
    <br />
    <span
      className="DetailText"
    >
      Longitude:
    </span>
    <input
      autoFocus={true}
      className="InputText"
      id="longitude"
    />
    <br />
    <span
      className="DetailText"
    >
      Population:
    </span>
    <input
      className="InputText"
      id="population"
      type="number"
    />
    <br />
    <br />
    <button
      className="btnText"
      onClick={[Function]}
    >
      Add City
    </button>
  </div>
</div>
`;
